### ISSUE TITLE
RLS Crashes

### ISSUE THREAD

**Issue by veremenko-y** [rust-lang-nursery/rls-vscode#67]
I have seen closed ticket for this, but it works ones and fails. Am I doing something wrong?

versions
```
$ cargo --version
cargo 0.21.0-nightly (534ce6862 2017-06-21)
$ rustup --version
rustup 1.5.0 (92d0d1e9e 2017-06-24)
$ rustc --version
rustc 1.20.0-nightly (859c3236e 2017-06-26)
```

settings.json
```
"rust.rustup": {
    "toolchain": "stable-x86_64-pc-windows-msvc",
    "nightlyToolchain": "nightly-x86_64-pc-windows-msvc"
},
"rust.mode": "rls",
"rust.rls": {
    "useRustfmt": true
}
```

Rust Log

```
DEBUG: Rustup: updateToolchains: this.toolchains=[{"channel":"stable","host":"x86_64-pc-windows-msvc","isDefault":false},{"channel":"nightly","host":"x86_64-pc-windows-msvc","isDefault":true}]
DEBUG: activate: processPossibleSetButMissingUserToolchain: toolchainKind=toolchain
DEBUG: activate: processPossibleSetButMissingUserToolchain: user toolchain is installed
DEBUG: Rustup: updateComponents(stable-x86_64-pc-windows-msvc): components=["cargo-x86_64-pc-windows-msvc (default)","rust-analysis-x86_64-pc-windows-msvc","rust-docs-x86_64-pc-windows-msvc (default)","rust-src (installed)","rust-std-aarch64-apple-ios","rust-std-aarch64-linux-android","rust-std-aarch64-unknown-fuchsia","rust-std-aarch64-unknown-linux-gnu","rust-std-arm-linux-androideabi","rust-std-arm-unknown-linux-gnueabi","rust-std-arm-unknown-linux-gnueabihf","rust-std-arm-unknown-linux-musleabi","rust-std-arm-unknown-linux-musleabihf","rust-std-armv7-apple-ios","rust-std-armv7-linux-androideabi","rust-std-armv7-unknown-linux-gnueabihf","rust-std-armv7-unknown-linux-musleabihf","rust-std-armv7s-apple-ios","rust-std-asmjs-unknown-emscripten","rust-std-i386-apple-ios","rust-std-i586-pc-windows-msvc","rust-std-i586-unknown-linux-gnu","rust-std-i686-apple-darwin","rust-std-i686-linux-android","rust-std-i686-pc-windows-gnu","rust-std-i686-pc-windows-msvc","rust-std-i686-unknown-freebsd","rust-std-i686-unknown-linux-gnu","rust-std-i686-unknown-linux-musl","rust-std-mips-unknown-linux-gnu","rust-std-mips-unknown-linux-musl","rust-std-mips64-unknown-linux-gnuabi64","rust-std-mips64el-unknown-linux-gnuabi64","rust-std-mipsel-unknown-linux-gnu","rust-std-mipsel-unknown-linux-musl","rust-std-powerpc-unknown-linux-gnu","rust-std-powerpc64-unknown-linux-gnu","rust-std-powerpc64le-unknown-linux-gnu","rust-std-s390x-unknown-linux-gnu","rust-std-sparc64-unknown-linux-gnu","rust-std-wasm32-unknown-emscripten","rust-std-x86_64-apple-darwin","rust-std-x86_64-apple-ios","rust-std-x86_64-linux-android","rust-std-x86_64-pc-windows-gnu","rust-std-x86_64-pc-windows-msvc (default)","rust-std-x86_64-rumprun-netbsd","rust-std-x86_64-unknown-freebsd","rust-std-x86_64-unknown-fuchsia","rust-std-x86_64-unknown-linux-gnu","rust-std-x86_64-unknown-linux-musl","rust-std-x86_64-unknown-netbsd","rustc-x86_64-pc-windows-msvc (default)",""]
DEBUG: RlsMode: start: enter
DEBUG: RlsMode: start: no RLS executable
DEBUG: RlsMode: start: processPossibleSetButMissingUserToolchain: toolchainKind=nightly toolchain
DEBUG: RlsMode: start: processPossibleSetButMissingUserToolchain: user nightly toolchain is installed
DEBUG: Rustup: updateComponents(nightly-x86_64-pc-windows-msvc): components=["cargo-x86_64-pc-windows-msvc (default)","rls-x86_64-pc-windows-msvc (installed)","rust-analysis-x86_64-pc-windows-msvc (installed)","rust-docs-x86_64-pc-windows-msvc (default)","rust-src (installed)","rust-std-aarch64-apple-ios","rust-std-aarch64-linux-android","rust-std-aarch64-unknown-fuchsia","rust-std-aarch64-unknown-linux-gnu","rust-std-arm-linux-androideabi","rust-std-arm-unknown-linux-gnueabi","rust-std-arm-unknown-linux-gnueabihf","rust-std-arm-unknown-linux-musleabi","rust-std-arm-unknown-linux-musleabihf","rust-std-armv7-apple-ios","rust-std-armv7-linux-androideabi","rust-std-armv7-unknown-linux-gnueabihf","rust-std-armv7-unknown-linux-musleabihf","rust-std-armv7s-apple-ios","rust-std-asmjs-unknown-emscripten","rust-std-i386-apple-ios","rust-std-i586-pc-windows-msvc","rust-std-i586-unknown-linux-gnu","rust-std-i686-apple-darwin","rust-std-i686-linux-android","rust-std-i686-pc-windows-gnu","rust-std-i686-pc-windows-msvc","rust-std-i686-unknown-freebsd","rust-std-i686-unknown-linux-gnu","rust-std-i686-unknown-linux-musl","rust-std-mips-unknown-linux-gnu","rust-std-mips-unknown-linux-musl","rust-std-mips64-unknown-linux-gnuabi64","rust-std-mips64el-unknown-linux-gnuabi64","rust-std-mipsel-unknown-linux-gnu","rust-std-mipsel-unknown-linux-musl","rust-std-powerpc-unknown-linux-gnu","rust-std-powerpc64-unknown-linux-gnu","rust-std-powerpc64le-unknown-linux-gnu","rust-std-s390x-unknown-linux-gnu","rust-std-sparc64-unknown-linux-gnu","rust-std-wasm32-unknown-emscripten","rust-std-x86_64-apple-darwin","rust-std-x86_64-apple-ios","rust-std-x86_64-linux-android","rust-std-x86_64-pc-windows-gnu","rust-std-x86_64-pc-windows-msvc (default)","rust-std-x86_64-rumprun-netbsd","rust-std-x86_64-unknown-freebsd","rust-std-x86_64-unknown-fuchsia","rust-std-x86_64-unknown-linux-gnu","rust-std-x86_64-unknown-linux-musl","rust-std-x86_64-unknown-netbsd","rustc-x86_64-pc-windows-msvc (default)",""]
DEBUG: RlsMode: processPossiblyMissingRlsComponents: RLS is installed
DEBUG: RlsMode: processPossiblyMissingRlsComponents: rust-analysis is installed
DEBUG: RlsMode: start: User decided to use rustfmt
DEBUG: RlsMode: start: rlsPath=rustup 
DEBUG: RlsMode: start: env={"RUST_SRC_PATH":"C:\\Users\\yaros\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\src\\rust\\src"} 
DEBUG: RlsMode: start: args=["run","nightly-x86_64-pc-windows-msvc","rls"] 
DEBUG: RlsMode: start: revealOutputChannelOn=3 
DEBUG: RlsMode: start: Language Client Manager: start
```

RLS Log
```
{"message":"cannot find value 'COMMAND_' in module 'cpu_commands'","code":{"code":"E0425","explanation":"\nAn unresolved name was used.\n\nErroneous code examples:\n\n'''compile_fail,E0425\nsomething_that_doesnt_exist::foo;\n// error: unresolved name 'something_that_doesnt_exist::foo'\n\n// or:\n\ntrait Foo {\n    fn bar() {\n        Self; // error: unresolved name 'Self'\n    }\n}\n\n// or:\n\nlet x = unknown_variable;  // error: unresolved name 'unknown_variable'\n'''\n\nPlease verify that the name wasn't misspelled and ensure that the\nidentifier being referred to is valid for the given situation. Example:\n\n'''\nenum something_that_does_exist {\n    Foo,\n}\n'''\n\nOr:\n\n'''\nmod something_that_does_exist {\n    pub static foo : i32 = 0i32;\n}\n\nsomething_that_does_exist::foo; // ok!\n'''\n\nOr:\n\n'''\nlet unknown_variable = 12u32;\nlet x = unknown_variable; // ok!\n'''\n\nIf the item is not defined in the current module, it must be imported using a\n'use' statement, like so:\n\n'''\n# mod foo { pub fn bar() {} }\n# fn main() {\nuse foo::bar;\nbar();\n# }\n'''\n\nIf the item you are importing is not defined in some super-module of the\ncurrent module, then it must also be declared as public (e.g., 'pub fn').\n"},"level":"error","spans":[{"file_name":"src\\main.rs","byte_start":78,"byte_end":86,"line_start":4,"line_end":4,"column_start":48,"column_end":56,"is_primary":true,"text":[{"text":"    println!(\"Hello, world! {}\", cpu_commands::COMMAND_);","highlight_start":48,"highlight_end":56}],"label":"not found in 'cpu_commands'","suggested_replacement":null,"expansion":null}],"children":[],"rendered":null}
Missing change, aborting. Found 2, expected Some(11)
[Error - 3:37:24 PM] Connection to server got closed. Server will not be restarted.
```

**Comment by josephlr** [comment]
Looks like #26 should be reopened and this issued merged in.
I can confirm this error reproduces for me with:
```
rustup 1.5.0 (92d0d1e9e 2017-06-24)
rustc 1.20.0-nightly (f590a44ce 2017-06-27)
cargo 0.21.0-nightly (abf01e1ed 2017-06-25)
```

I get an output below with various numbers on crash. No additional output in the Rust logging
```
Missing change, aborting. Found 13, expected Some(15)
[Error - 10:16:20 PM] Connection to server got closed. Server will not be restarted.
```

Settings:
```json
"rust.rustup": {
    "toolchain": "nightly-x86_64-apple-darwin",
    "nightlyToolchain": "nightly-x86_64-apple-darwin"
 },
"rust.mode": "rls",
"rust.rls": {
    "useRustfmt": true
}
```

**Comment by josephlr** [comment]
Never mind, the fix to `rls` has not been pushed to rustup yet. `rls --version` should still show 6/20 (before the fix for rust-lang-nursery/rls#312). You can either wait for the fix to be pushed out soon, or follow the [insructions](https://github.com/rust-lang-nursery/rls/blob/master/contributing.md) to build `rls` from source and then plug that into the extension.

**Comment by veremenko-y** [comment]
Hi @josephlr. Thank you very much. I switched to legacy mode for now, I will wait. Anyway I'm just learning from very beginning now.

I will close the issue.
